### PR TITLE
Detailed metrics views for each http endpoint

### PR DESCRIPTION
**Pull request by camelCasedAditya**
Each http endpoint now has a detailed view which the user can access by navigating to the endpoint they want on the table and then clicking enter. In the detailed view there is metrics specific to that endpoint and a bar chart showing the number of requests under X latency in ms. There is also a sparkline graph chart that shows the number of requests per second over the past ___ seconds. The user can view a longer timeframe by expanding the console window horizontally. In the main view all endpoints currently receiving requests are colored purple, ingest and consumption endpoints colored green (not getting requests), and other endpoints colored cyan (not getting requests).

Detailed view UI:

In iterm code terminal:
image

In normal mac terminal:
image

Example of highlighted endpoints
image

### CODE REVIEW

File: apps/framework-cli/src/cli/routines/metrics_console/run_console/app.rs [hunk 1]
```diff
@@ -1,26 +1,39 @@
...
+    pub path_detailed_data: Option<Vec<HistogramCount>>,
+    pub path_requests_per_sec: HashMap<String, f64>,
+    pub requests_per_sec_vec: HashMap<String, Vec<u64>>,
 }
 
 impl Default for App {
     fn default() -> Self {
         Self {
+            viewport: Rect::new(0, 0, 0, 0),
+            state: "main".to_string(),
```
Comment (callicles) [inline apps/framework-cli/src/cli/routines/metrics_console/run_console/app.rs]: What's this state? which panel this is on? if that's the case should this be an enum instead?

File: apps/framework-cli/src/cli/routines/metrics_console/run_console/ui.rs [hunk 2]
```diff
@@ -8,116 +8,357 @@ use ratatui::{prelude::*, widgets::*};
 
 use crate::cli::routines::metrics_console::run_console::app::App;
 
-const INFO_TEXT: &str = "(q) quit | (↑) move up | (↓) move down";
+const INFO_TEXT: &str = "(q) quit | (↑) move up | (↓) move down | (enter) view endpoint details";
 
 /// Renders the user interface widgets.
 pub fn render(app: &mut App, frame: &mut Frame) {
```
Comment (callicles) [inline apps/framework-cli/src/cli/routines/metrics_console/run_console/ui.rs:15]: This is a huge method - could you break it down in smaller ones?

File: apps/framework-cli/src/cli/routines/metrics_console/run_console/app.rs [hunk 3]
```diff
@@ -53,7 +72,43 @@ impl App {
         }
     }
 
-    pub fn req_per_sec(&mut self, new_total_requests: f64) {
+    pub fn req_per_sec(&mut self, new_total_requests: f64, path_metrics: &Vec<PathMetricsData>) {
         self.requests_per_sec = new_total_requests - self.total_requests;
+
+        // Initializes variables and vec for each path in summary to keep unwraps in ui.rs safe
```
Comment (callicles) [inline apps/framework-cli/src/cli/routines/metrics_console/run_console/app.rs:83]: In general, convention across files will create bugs when someone else comes and changes things. People should be able to assume that they can change things locally to a function and that it won't break things in other places. Here this convention breaks that principle.

It would have been ok, in the same function to have something that assumes you can unwrap and makes sure that's safe. But that definitely should not be done cross files. That's what the types are here for.

**Review by callicles — CHANGES_REQUESTED** [review]
Some small changes to make it good to merge!

File: apps/framework-cli/src/cli/local_webserver.rs [hunk 4]
```diff
@@ -492,7 +492,19 @@ async fn router(
             .body(Full::new(Bytes::from("no match"))),
     };
 
-    if metrics_path.clone().into_os_string().to_str().unwrap() != "metrics" {
+    if metrics_path
+        .clone()
+        .into_os_string()
```
Comment (phiSgr) [inline apps/framework-cli/src/cli/local_webserver.rs]: seems like we don't need these two lines?

File: apps/framework-cli/src/cli/local_webserver.rs [hunk 5]
```diff
@@ -492,7 +492,19 @@ async fn router(
...
+        .to_str()
+        .unwrap()
+        .contains("ingest/")
```
Comment (phiSgr) [inline apps/framework-cli/src/cli/local_webserver.rs]: `starts_with` should be a more performant and robust check

File: apps/framework-cli/src/cli/routines/metrics_console/run_console/app.rs [hunk 6]
```diff
@@ -53,7 +72,43 @@ impl App {
...
+            }
+        }
+
+        for path in &self.summary {
+            for item in path_metrics {
+                if item.path == path.path {
+                    self.path_requests_per_sec
+                        .insert(path.path.clone(), item.request_count - path.request_count);
+                }
+            }
```
Comment (phiSgr) [inline apps/framework-cli/src/cli/routines/metrics_console/run_console/app.rs:100]: this block seems unrelated to the outer loop.

**Review by callicles — APPROVED** [review]
